### PR TITLE
Improve shell script detection

### PR DESCRIPTION
**Pull request by jamacku**

### CODE REVIEW

File: scripts/run-shellcheck.sh [hunk 1]
```diff
@@ -2,8 +2,8 @@
 export LC_ALL="C"
 
 find "$@" -type f -print0 \
-    | xargs -0 sh -c 'for i in "$@"; do { printf "%s\n" "$i" | grep "\\.sh\$" \
-        || head -n1 "$i" | grep --text -Ev "(atf-|tcl|wi)sh\$" | grep --text "^#!.*sh\$"
+    | xargs -0 sh -c 'for i in "$@"; do { printf "%s\n" "$i" | grep -E "^.*\\.(ash|bash|bats|dash|ksh|sh)$" \
```
Comment (jamacku) [inline scripts/run-shellcheck.sh:5]: @kdudka Isn't there missing `\`?

```suggestion
    | xargs -0 sh -c 'for i in "$@"; do { printf "%s\\n" "$i" | grep -E "^.*\\.(ash|bash|bats|dash|ksh|sh)$" \
```

Comment (kdudka) [reply scripts/run-shellcheck.sh:5]: As discussed in person, both the variants seem to work equally but I will add the backslash for clarity.

**Review by kdudka — APPROVED** [review]
Merging, thanks for the contribution!  I will open another pull request to make this code easier to maintain...

File: .github/workflows/differential-shellcheck.yml [hunk 2]
```diff
@@ -22,7 +23,15 @@ jobs:
...
           fetch-depth: 0
 
-      - name: Differential ShellCheck
+      - id: ShellCheck
+        name: Differential ShellCheck
         uses: redhat-plumbers-in-action/differential-shellcheck@v4
         with:
           token: ${{ secrets.GITHUB_TOKEN }}
+
+      - if: ${{ always() }}
```
Comment (lzaoral) [inline .github/workflows/differential-shellcheck.yml:32]: Isn't this a bit redundant? 🤔

Comment (lzaoral) [reply .github/workflows/differential-shellcheck.yml:32]: Never mind, I will run even when the previous step fails, right?

Comment (jamacku) [reply .github/workflows/differential-shellcheck.yml:32]: correct :+1:

**Pull request merged**